File: informix/src/test/java/com/dbx/agent/informix/InformixAgentExecutionTest.java
package com.dbx.agent.informix;

import com.dbx.agent.DatabaseAgent;
import com.dbx.agent.test.JdbcFakeExecutionBehaviorTest;

class InformixAgentExecutionTest extends JdbcFakeExecutionBehaviorTest {
    @Override
    protected DatabaseAgent createAgent() {
        return new InformixAgent();
    }

    @Override
    protected String resultSetSql() {
        return "EXECUTE PROCEDURE sample_proc()";
    }
}

File: informix/src/test/kotlin/com/dbx/agent/informix/InformixAgentExecutionTest.kt
package com.dbx.agent.informix

import com.dbx.agent.test.JdbcAgentFake
import com.dbx.agent.test.setPrivateConnection
import kotlin.test.Test
import kotlin.test.assertEquals

class InformixAgentExecutionTest {
    @Test
    fun `executes non select statements that return result sets`() {
        val agent = InformixAgent()
        setPrivateConnection(agent, JdbcAgentFake.connection())

        val result = agent.executeQuery("EXECUTE PROCEDURE sample_proc()", null)

        assertEquals(listOf("VALUE"), result.columns)
        assertEquals(listOf(listOf("row-value")), result.rows)
        assertEquals(listOf("execute"), JdbcAgentFake.calls)
    }
}
